### ISSUE TITLE
Add ability to preempt nuget search with local folder ('NuGetPrefetchFolder')

### ISSUE THREAD

**Issue by JanKrivanek** [NuGet/Home#12515]
### NuGet Product(s) Involved

NuGet SDK

### The Elevator Pitch

Idea brainstormed by @zivkan

For manuall or automated ways of testing multi-project changes, that are tight together via PackageReference (e.g. project A, references project B as nuget, while developer needs to perform orchestrated change to both) the existing functionality of `NuGetFallbackFolder` might be tweaked and reused for easy testing of validity of the nuget package without the need to 'hacking' nuget cache.
 * a nuget of B would be build
 * added to relative `NuGetPrefetchFolder` (that one would not get synced to global cache - same as `NuGetFallbackFolder`)
 * restore would check the `NuGetPrefetchFolder` first and get package from there if available
 * other unrelated project would get the B from global cache (skiping the `NuGetPrefetchFolder`)

If functionality of `NuGetFallbackFolder` would be reused, then possibly a CLI support for equivalent of `nuget add` would need to be added - in order to be able to unzip and push a local nuget to local folder in v3 layout.

Other question is the location of `NuGetPrefetchFolder` - is it fixed or configurable via packages.config? It needs to be per 'build-bundle', not global to the machine - as we want to avoid 'poisoning' of nugets for unrelated builds

**Comment by zivkan** [comment]
This is effectively a design proposal for https://github.com/NuGet/Home/issues/6579

**Comment by gioce90** [comment]
> For manuall or automated ways of testing multi-project changes [...] might be tweaked and reused for easy testing of validity of the nuget package without the need to 'hacking' nuget cache.

I agree. I’ve faced similar issues in the past when working with NuGet packages. As I mentioned [here](https://github.com/NuGet/Home/issues/6579#issuecomment-2476832378).
However, in the meantime I’ve decided to make my solution public by releasing a package called NuJet (https://github.com/gioce90/NuJet). It enhance the inner-loop development and testing process for NuGet packages, especially in local.

Feel free to check it out—I’d love to hear your feedback!
